src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBaseSourceConfig.java

```diff
@@ -269,4 +277,10 @@ private int parseUnitValue(String propertyName, String value) {
   private ZonedDateTime parseDatetime(String datetime) throws DateTimeParseException {
     return StringUtils.isBlank(datetime) ? null : ZonedDateTime.parse(datetime, DateTimeFormatter.ISO_DATE_TIME);
   }
+
+  public String getOperation() {
```

Comment:
add code in validate() to check the operation is either query or queryAll. Add failure to failure collector if the operation is invalid

Reply:
so add similar logic to whats being done with validateFilters()?

Reply:
yes. convert operation string to enum. If that fails (exception is thrown), then add a failure to the failure colllector.